### ISSUE TITLE
Libgcrypt warning: missing initialization when mounting

### ISSUE THREAD

**Issue by acasademont** [s3fs-fuse/s3fs-fuse#524]
#### Additional Information
_The following information is very important in order to help us to help you. Omission of the following details may delay your support request or receive no attention at all._

- Version of s3fs being used (s3fs --version)
- 1.80 (Amazon Simple Storage Service File System V1.80(commit:unknown) with GnuTLS(gcrypt))

- Version of fuse being used (pkg-config --modversion fuse)
 - 2.9.3-15+deb8u2

- System information (uname -a)
 - Linux xxxx 3.16.0-4-amd64 #1 SMP Debian 3.16.36-1+deb8u2 (2016-10-19) x86_64 GNU/Linux

- Distro (cat /etc/issue)
 - Debian 8 Jessie

- /etc/fstab entry (if applicable): 
```
xxxx /mnt/media fuse.s3fs _netdev,allow_other,use_cache=/tmp,endpoint=eu-west-1,nonempty 0 0
```
- s3fs syslog messages (grep s3fs /var/log/syslog, or s3fs outputs)  
_if you execute s3fs with dbglevel, curldbg option, you can get detail debug messages_
```
Jan 23 10:38:37 xxxx s3fs[27704]: s3fs.cpp:s3fs_init(3334): init v1.80(commit:unknown) with GnuTLS(gcrypt)
Jan 23 10:38:37 xxxx s3fs[27704]: Libgcrypt warning: missing initialization - please fix the application
```
#### Details about issue

When I mount I see this warning `s3fs[28054]: Libgcrypt warning: missing initialization - please fix the application`. Is this something I should worry about?

**Comment by ggtakec** [comment]
@acasademont 
I confirmed that s3fs initializes the gnutls library before calling the curl library.
gnutls is initialized by gnutls_global_init() function and gcry_check_version() function for initializing libgcry is called from this gnutls_global_init().
It seems that there is nothing wrong with it.

However, s3fs initialized SSL libraries twice at foreground mode, it had fixed by #546.

If you were running s3fs in the foreground, warnings may not be issued in the code of the new master branch.
(However, the warning may be continuing.)

In order to solve it, I think that we need to know the version number of gnutls/libgcrypt you are using.
And please try using s3fs with the latest gnutls/libgcrypt.

Regards,

**Comment by ggtakec** [comment]
@wjt Thank you for your help.
@acasademont , I am sorry for the delay in response. I've merged the fixed code, so it will be helpful if you check with the new code.
